test/utils/create_resources.go

```diff
@@ -53,15 +53,24 @@ func RetryWithExponentialBackOff(fn wait.ConditionFunc) error {
 	return wait.ExponentialBackoff(backoff, fn)
 }
 
-func CreatePodWithRetries(c clientset.Interface, namespace string, obj *v1.Pod) error {
+func CreatePodWithRetries(c clientset.Interface, namespace string, obj *v1.Pod, opts ...Option) error {
```

Comment:
We'd better to fix all other methods since this is problem for all the `CreatexxxWithRetries` method, not just Pod creation, actually, the retry is not effective at all.

Reply:
I could be ok with retrying this error, but retrying on timeouts in a generic helper like this is not correct.

You could update a perf test to retry, but not a generic helper that people will assume is safe and reasonable for e2e tests.

Reply:
Note that this PR doesn't change any default behavior - it only provides extra options for client to customize the behavior upon errors. And to be conservative, I'd like to start with `CreatePod()` only.

Reply:
> Note that this PR doesn't change any default behavior - it only provides extra options for client to customize the behavior upon errors. And to be conservative, I'd like to start with `CreatePod()` only.

I don't think this option should be available to general test authors using these utilities.  We've seen abuse (inconsistent abuse even), in the past.  Finding a way to wrap  this in ` test/integration/scheduler_perf` is preferable.

Reply:
+1 on doing the reties in scheduler_perf